### PR TITLE
refactor(panel, shell, tree-item): Add keys to conditionally rendered slots.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** N/A

## Summary

refactor(panel, shell, tree-item): Add keys to conditionally rendered slots.

### CODE REVIEW

File: src/components/calcite-shell/calcite-shell.tsx [hunk 1]
```diff
@@ -67,22 +67,27 @@ export class CalciteShell implements ConditionalSlotComponent {
...
+            {defaultSlotNode}
           </div>,
-          <slot name={SLOTS.centerRow} />
+          centerRowSlotNode
         ]
       : [
-          <div class={CSS.content}>
-            <slot />
-            <slot name={SLOTS.centerRow} />
+          <div class={CSS.content} key={contentContainerKey}>
```
Comment (jcfranco) [inline src/components/calcite-shell/calcite-shell.tsx:88]: Before merging, can you confirm that the CSS looks correct when `contentBehind` is toggled?

File: src/components/calcite-shell/calcite-shell.tsx [hunk 2]
```diff
@@ -67,22 +67,27 @@ export class CalciteShell implements ConditionalSlotComponent {
...
+    const defaultSlotNode: VNode = <slot key="default-slot" />;
+    const centerRowSlotNode: VNode = <slot key="center-row-slot" name={SLOTS.centerRow} />;
+    const contentContainerKey = "content-container";
+
     const content = !!this.contentBehind
       ? [
           <div
             class={{
               [CSS.content]: true,
               [CSS.contentBehind]: !!this.contentBehind
```
Comment (jcfranco) [inline src/components/calcite-shell/calcite-shell.tsx:79]: Sidebar/nitpick: this is always `true` here.

**Review by jcfranco — APPROVED** [review]
👍🏼

File: src/components/calcite-shell/calcite-shell.tsx (hunk 1 above)
Comment (driskull) [reply src/components/calcite-shell/calcite-shell.tsx:88]: Looks good!

**Pull request merged**